File: android/src/test/java/apitests/java/SubscriberAttributesApiTests.java
package apitests.java;

import com.revenuecat.purchases.hybridcommon.SubscriberAttributesKt;

import java.util.Map;

@SuppressWarnings("unused")
class SubscriberAttributesApiTests {
    // region Attribution IDs

    private void checkCollectDeviceIdentifiers() {
        SubscriberAttributesKt.collectDeviceIdentifiers();
    }

    private void checkSetAdjustID(String id) {
        SubscriberAttributesKt.setAdjustID(id);
    }

    private void checkSetAppsflyerId(String id) {
        SubscriberAttributesKt.setAppsflyerID(id);
    }

    private void checkSetFBAnonymousID(String id) {
        SubscriberAttributesKt.setFBAnonymousID(id);
    }

    private void checkSetMparticleID(String id) {
        SubscriberAttributesKt.setMparticleID(id);
    }

    private void checkSetOnesignalID(String id) {
        SubscriberAttributesKt.setOnesignalID(id);
    }

    private void checkSetAirshipChannelID(String id) {
        SubscriberAttributesKt.setAirshipChannelID(id);
    }

    // endregion
    // region Campaign parameters

    private void checkSetMediaSource(String mediaSource) {
        SubscriberAttributesKt.setMediaSource(mediaSource);
    }

    private void checkSetCampaign(String campaign) {
        SubscriberAttributesKt.setCampaign(campaign);
    }

    private void checkSetAdGroup(String adGroup) {
        SubscriberAttributesKt.setAdGroup(adGroup);
    }

    private void checkSetAd(String ad) {
        SubscriberAttributesKt.setAd(ad);
    }

    private void checkSetKeyword(String keyword) {
        SubscriberAttributesKt.setKeyword(keyword);
    }

    private void checkSetCreative(String creative) {
        SubscriberAttributesKt.setCreative(creative);
    }

    // endregion
    // region subscriber attributes

    private void checkSetAttributes(Map<String, String> attributes) {
        SubscriberAttributesKt.setAttributes(attributes);
    }

    private void checkSetEmail(String email) {
        SubscriberAttributesKt.setEmail(email);
    }

    private void checkSetPhoneNumber(String phoneNumber) {
        SubscriberAttributesKt.setPhoneNumber(phoneNumber);
    }

    private void checkSetDisplayName(String displayName) {
        SubscriberAttributesKt.setDisplayName(displayName);
    }

    private void checkSetPushToken(String pushToken) {
        SubscriberAttributesKt.setPushToken(pushToken);
    }

    // endregion
}

File: android/src/test/java/apitests/kotlin/com/revenuecat/purchases/hybridcommon/SubscriberAttributesApiTests.kt
package apitests.kotlin.com.revenuecat.purchases.hybridcommon

import com.revenuecat.purchases.hybridcommon.*

@Suppress("unused")
private class SubscriberAttributesApiTests {

    // region Attribution IDs

    fun checkCollectDeviceIdentifiers() {
        collectDeviceIdentifiers()
    }

    fun checkSetAdjustID(id: String?) {
        setAdjustID(id)
    }

    fun checkSetAppsflyerID(id: String?) {
        setAppsflyerID(id)
    }

    fun checkSetFBAnonymousID(id: String?) {
        setFBAnonymousID(id)
    }

    fun checkSetMparticleID(id: String?) {
        setMparticleID(id)
    }

    fun checkSetOnesignalID(id: String?) {
        setOnesignalID(id)
    }

    fun checkSetAirshipChannelID(id: String?) {
        setAirshipChannelID(id)
    }

    // endregion
    // region Campaign parameters

    fun checkSetMediaSource(mediaSource: String?) {
        setMediaSource(mediaSource)
    }

    fun checkSetCampaign(campaign: String?) {
        setCampaign(campaign)
    }

    fun checkSetAdGroup(adGroup: String?) {
        setAdGroup(adGroup)
    }

    fun checkSetAd(ad: String?) {
        setAd(ad)
    }

    fun checkSetKeyword(keyword: String?) {
        setKeyword(keyword)
    }

    fun checkSetCreative(creative: String?) {
        setCreative(creative)
    }

    // endregion
    // region subscriber attributes

    fun checkSetAttributes(attributes: Map<String, String?>) {
        setAttributes(attributes)
    }

    fun checkSetEmail(email: String?) {
        setEmail(email)
    }

    fun checkSetPhoneNumber(phoneNumber: String?) {
        setPhoneNumber(phoneNumber)
    }

    fun checkSetDisplayName(displayName: String?) {
        setDisplayName(displayName)
    }

    fun checkSetPushToken(fcmToken: String?) {
        setPushToken(fcmToken)
    }

    // endregion
}
